sotabencheval/language_modelling/wikitext.py

```diff
@@ -0,0 +1,132 @@
+import os
+from itertools import islice
+from enum import Enum
+from pathlib import Path
+
+import numpy as np
+
+from sotabenchapi.check import in_check_mode
+from sotabenchapi.client import Client
+from sotabenchapi.core import BenchmarkResult, check_inputs
+from sotabencheval.core import BaseEvaluator
+from sotabencheval.utils import calculate_batch_hash, extract_archive, change_root_if_server, is_server
+
+def to_numpy(*args):
+    def convert(a):
+        if hasattr(a, 'cpu') and hasattr(a, 'numpy'):
+            return a.cpu().numpy()
+        if isinstance(a, list):
+            return np.array(a)
+        return a 
+    return [convert(a) for a in args]
+
+class WikiTextDataset(Enum):
+    WikiText103 = ('WikiText-103', 245569, 267735)
+    WikiText2 = ('WikiText-2', 245569, 33278)
+    
+    def __init__(self, pwc_name, testset_size, vocab_size):
+        self.pwc_name = pwc_name
+        self.testset_size = testset_size
+        self.vocab_size = vocab_size
+    
+    def get_path(self, local_root, local_unzip=False):
+        root = Path(change_root_if_server(root=local_root,
```

Comment:
what does this method do? where is it used?  docstrings

Reply:
it is the methods that convience method to extract the dataset `WikText103Evaluator.dataset.get_path`. Will add the docs strings.

Reply:
ty - and also a step-by-step example where this is used